lib/preload/ndb/httpMonkeyPatching.js

```diff
@@ -0,0 +1,153 @@
+const zlib = require('zlib');
```

Comment:
We need to use another way to inject this script, and another channel for inspector node process to speak with DevTools frontend. We already have channel between node process and frontend - devtools protocol.

To inject this script please do following:
- add httpMonkeyPatchingSource method to backend.js, this method returns source of this file.
- `Ndb.NodeProcessManager.detected` method gets source of this script and inject it to the page using following snippet:
```javascript
target.runtimeAgent().invoke_evaluate({
  expression: await Ndb.backend.httpMonkeyPatchingSource()
});
```

After these steps we can inject monkey patching script to any inspected process. Second step is how we can build a channel. We can use `Runtime.evaluate` with `awaitPromise: true` flag to build a channel. Monkey patching script gets following code:
```javascript
let messages = [];
let messageAdded = null;

// this function is our instrumentation, to report anything to frontend - call it instead of process.send
function reportMessage(message) {
  messages.push(message);
  if (messageAdded) {
    setTimeout(messageAdded, 0);
    messageAdded = null;
  }
}

// this function should be called from frontend in the loop using `target.runtimeAgent().invoke_evaluate`
process._getNetworkMessages = async function() {
    if (!messages.length)
      await new Promise(resolve => messageAdded = resolve);
    return messages.splice(0);
  }
}
```

Frontend calls in the loop following code:
```javascript
while (true) {
  const messages = await target.runtimeAgent().invoke_evaluate({
    expression: 'process._getNetworkMessages()', awaitPromise: true
  });
  // ... process these messages ...
}
```

Feel free to ask any questions! At the same time we can merge your pull request and I will refactor it.